### PR TITLE
Enrichs docs and tests regarding empty strings used in matchers (#20521).

### PR DESCRIPTION
**Pull request by rbudzko**
Please, see comment in #20521.

### CODE REVIEW

**Comment by ktoso** [comment]
OK TO TEST

File: akka-docs/rst/java/http/routing-dsl/directives/path-directives/path.rst [hunk 1]
```diff
@@ -24,6 +24,10 @@ a ``path`` directive will always be empty).
 Depending on the type of its ``PathMatcher`` argument the ``path`` directive extracts zero or more values from the URI.
 If the match fails the request is rejected with an :ref:`empty rejection set <empty rejections>`.
 
+.. note:: The empty string (also called empty word or identity) is a **neutral element** of string concatenation operation,
+ so it will match everything, but remember that ``path`` requires whole remaining path being matched, so (``/``) will succeed
```
Comment (johanandren) [inline akka-docs/rst/java/http/routing-dsl/directives/path-directives/path.rst]: I'm pretty sure this needs to be indented, please generate docs using `akka-docs/sphinx:generateHtml` and take a look at the generated html docs. 

( You can find instructions on what needs to be installed here: http://doc.akka.io/docs/akka/snapshot/dev/documentation.html )

**Comment by johanandren** [comment]
LGTM except the indentation

File: akka-docs/rst/java/http/routing-dsl/directives/path-directives/path.rst (hunk 1 above)
Comment (rbudzko) [inline akka-docs/rst/java/http/routing-dsl/directives/path-directives/path.rst]: @johanandren, it is indented by one space. You are right about each line following ``::note`` requiring this additional space to be rendered inside of blue box. I've verified it based on html output of validatePullRequest on full project.

For a very stupid reason cannot revalidate before Friday :-( ...

Comment (johanandren) [inline akka-docs/rst/java/http/routing-dsl/directives/path-directives/path.rst]: Ah, sorry, I couldn't see that it was already indented (additionally I thought the requirement was 3 spaces, but 1 may very well be enough)

**Comment by ktoso** [comment]
Correct findings and good docs :)
Thanks for the help @rbudzko !

**Pull request merged**